airbyte-integrations/connectors/source-marketo/source_marketo/utils.py

```diff
@@ -44,29 +44,30 @@ def clean_string(string: str) -> str:
 
 
 def format_value(value, schema):
-    if not isinstance(schema["type"], list):
-        field_type = [schema["type"]]
-    else:
-        field_type = schema["type"]
+    """Need use try...except construction for condition
+    when we get unexpected type from API"""
 
-    if value in [None, "", "null"]:
-        return None
-    elif "integer" in field_type:
-        if isinstance(value, int):
-            return value
+    try:
+        if not isinstance(schema["type"], list):
+            field_type = [schema["type"]]
+        else:
+            field_type = schema["type"]
 
-        # Custom Marketo percent type fields can have decimals, so we drop them
-        decimal_index = value.find(".")
-        if decimal_index > 0:
-            value = value[:decimal_index]
-        return int(value)
-    elif "string" in field_type:
-        return str(value)
-    elif "number" in field_type:
-        return float(value)
-    elif "boolean" in field_type:
-        if isinstance(value, bool):
-            return value
-        return value.lower() == "true"
+        if value in [None, "", "null"]:
+            return None
+        elif "integer" in field_type:
+            if isinstance(value, int):
```

Comment:
int(1) = 1 so I think we can elminate the if condition

Reply:
Updated